test/compilable/imports/a14528.d

```diff
@@ -1,6 +0,0 @@
-module imports.a14528;
-
-void foo(alias f)()
-{
-    f();
```

Comment:
Why would this work? It doesn't make any sense. If the function is protected, then it should not be callable from outside the package. Why does an alias parameter bypass that?

Reply:
The symbol should not be visible from this module - which is already rejected. But passing by `alias` is explicitly done to carry the dependency from one module to anotherand hence should be checked based location where the template is instantiated?

Reply:
What do you mean is already rejected? This was a compilable test, it did pass compilation.

You mean, at `foo!func()` we should check whether foo's body may access `f()`? This would depend on where the template instance is placed.

Reply:
> What do you mean is already rejected? This was a compilable test, it did pass compilation.

`imports/a14528.d` cannot access `C.func` directly?

> You mean, at foo!func() we should check whether foo's body may access f()? This would depend on where the template instance is placed.

No, when encountering ` foo!func()` it should check whether `func` is visible in the current scope (`C.test`)

Reply:
Oh, scratch that. I understand your point now. If `foo` is instantiated in the same module as `f`, then it should be ok.

Reply:
> imports/a14528.d cannot access C.func directly?

Nope, they should be in the same package.

> No, when encountering foo!func() it should check whether func is visible in the current scope (C.test)

Yes, that is the case, however, when inside the template instance the alias parameter has already been substituted with the aliased symbol. So the call to `f` is already translated as a call to `func`. At that point, it is difficult to undertand whether the call symbol was aliased or not.

Reply:
> At that point, it is difficult to undertand whether the call symbol was aliased or not.

Doesn't matter, the current behaviour is a breaking change that affects a IMO valid usage for `alias` parameters.
It's consistent with the possibility to create `public alias foo = bar` for `private XY bar`.

Reply:
> Yes, that is the case, however, when inside the template instance the alias parameter has already been substituted with the aliased symbol. So the call to `f` is already translated as a call to `func`. At that point, it is difficult to undertand whether the call symbol was aliased or not.

I hit the same wall when I tried to resolve this bug months ago, the info from the alias is lost.
I think a cheap solution could be to add a field to Dsymbol (e.g. `AliasDeclaration fromAlias;`), then any symbol that comes from an alias to private is copied and has this field set.

Reply:
Yeah, but then you would need to an array of such dsymbols as you may have multiple aliases to the same symbol (I don't know if it makes sense in practice to do so, however, the language allows it).

Reply:
> Also, how would you know, once the alias is substituted, if the user used the alias or the actual dsymbol?

You have a copy of the original symbol.

But well, maybe isn't the best idea, the issue is fresh on your mind not mine.

Reply:
This just continue to be compilable. The alias works as if you passed the address of the protected function to the callee.

Reply:
Unfortunately, I don't see a simple way to implement public aliases to private symbols.